distributed/tests/test_scheduler.py

```diff
@@ -2849,6 +2860,10 @@ async def test_get_worker_monitor_info(s, a, b):
         assert res[w.address]["last_time"] is not None
 
 
+@pytest.mark.skipif(
```

Comment:
I think we need to preserve this kind of behavior. I don't care if it's actually round robin or not but if I submit three tasks and there are two workers, both workers should have work.

What's the plan to fix this?

Reply:
> if I submit three tasks and there are two workers, both workers should have work

That's not what this is testing. What you're describing above already will happen. This PR in fact _improves_ that case. See https://github.com/dask/distributed/issues/7197: if you set worker-saturation to 1.1 right now, `test_wait_first_completed` will fail because 2 tasks get assigned to the 1-threaded worker, and one task is assigned to the 2-threaded worker, because they're using the old code path.

This test is testing that if you submit a task to a completely empty cluster, wait for it to complete, release it, then submit another task, that you'll get different workers each time. That's a different and much more niche case.

Approaches for this:
* https://github.com/dask/distributed/pull/7222
* https://github.com/dask/distributed/pull/7248

Reply:
Please delete the test